src/client/inputhandler.h

```diff
@@ -192,6 +211,14 @@ class RealInputHandler : public InputHandler
 	{
 		return m_receiver->WasKeyDown(keyCode);
 	}
+	virtual void listenForKey(const KeyPress &keyCode)
```

Comment:
They are virtual because they get overridden. If they aren't virtual, and I call ->listenForKey, it uses the original version from InputHandler. see also http://stackoverflow.com/a/2391781